src/Microsoft.PowerShell.Commands.Utility/commands/utility/Write-Object.cs

```diff
@@ -55,10 +55,8 @@ protected override void ProcessRecord()
             {
                 enumerate = false;
             }
-            foreach (PSObject inputObject in _inputObjects) // compensate for ValueFromRemainingArguments
-            {
-                WriteObject(inputObject, enumerate);
-            }
+
+            WriteObject(_inputObjects, enumerate);
```

Comment:
Man, what a mess.  Here's some of Write-Output's behavior today:

```
 write-output -NoEnumerate a b c | % { $_.GetType().FullName; $_.Count; $_ -join '/' }
System.String
1
a
System.String
1
b
System.String
1
c

 write-output -NoEnumerate a,b,c | % { $_.GetType().FullName; $_.Count; $_ -join '/' }
System.Object[]
3
a/b/c
```

This is a symptom of the flawed binding logic I changed (assuming the first call to BindParameter would fail when the `a,b,c` syntax was used), and the workaround that was created for it in Write-Output.  Whatever the correct behavior is, there should never be a difference between these two calls when ValueFromRemainingArguments is in effect:

```
Do-Something a b c
Do-Something a,b,c
```

The values bound to the parameter should be identical in both cases.